### PR TITLE
Refactor renderers

### PR DESCRIPTION
**Pull request by loevgaard**
This PR will change the return type on renderers. This was done to allow to show the toolbar even though elements are cached

### CODE REVIEW

**Comment by loevgaard** [comment]
Sadly I just discovered that it doesn't add nested elements to the element stack: 

image

I will have to think about how this can be fixed :|

File: src/Model/Block.php [hunk 1]
```diff
@@ -40,6 +40,11 @@ public function getIdentifier(): string
         return sprintf('sscms-block-%s', (string) $this->getCode());
     }
 
+    public function getType(): string
+    {
+        return 'block';
```
Comment (Roshyo) [inline src/Model/Block.php]: Why not using the constants ?

File: src/Model/Carousel.php [hunk 2]
```diff
@@ -41,6 +41,11 @@ public function getId(): ?int
         return $this->id;
     }
 
+    public function getType(): string
+    {
+        return 'carousel';
```
Comment (Roshyo) [inline src/Model/Carousel.php]: Why not using the constants ?

File: src/Model/View.php [hunk 3]
```diff
@@ -37,6 +37,11 @@ public function getIdentifier(): string
         return sprintf('sscms-view-%s', (string) $this->getCode());
     }
 
+    public function getType(): string
+    {
+        return 'view';
```
Comment (Roshyo) [inline src/Model/View.php]: Why not using the constants ?

File: src/Renderer/ElementId.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    public function isCarousel(): bool
+    {
+        return ElementInterface::TYPE_CAROUSEL === $this->type;
+    }
+
+    public function isView(): bool
+    {
+        return ElementInterface::TYPE_VIEW === $this->type;
+    }
```
Comment (Roshyo) [inline src/Renderer/ElementId.php:57]: Missing the `isPage` method ?

File: src/Resources/views/view/non_existing_template.html.twig [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<div class="sscms-view sscms-view-non-existing-template sscms-view-{{ code }}" style="background-color: rgba(202,0,2,0.25)">
+    The template <code>{{ code }}</code> does not exist. Please go to
+    <a href="{{ path('setono_sylius_cms_admin_template_create', {'code': code}) }}">/admin</a> to create this template
```
Comment (Roshyo) [inline src/Resources/views/view/non_existing_template.html.twig:4]: Shouldn't this be translated ?

File: src/Stack/ElementStack.php [hunk 6]
```diff
@@ -31,22 +34,22 @@ public function hasElements(): bool
...
     }
 
     public function getViews(): array
     {
-        return array_filter($this->elements, static function (ElementInterface $element): bool {
-            return $element instanceof ViewInterface;
+        return array_filter($this->elements, static function (ElementId $element): bool {
+            return $element->isView();
         });
     }
```
Comment (Roshyo) [inline src/Stack/ElementStack.php:54]: missing getPages() method ?

**Review by Roshyo — CHANGES_REQUESTED** [review]

File: src/Model/Block.php (hunk 1 above)
Comment (loevgaard) [reply src/Model/Block.php]: Haha, no idea. Newbie me

File: src/Resources/views/view/non_existing_template.html.twig (hunk 5 above)
Comment (loevgaard) [reply src/Resources/views/view/non_existing_template.html.twig:4]: Yes!

File: src/Stack/ElementStack.php (hunk 6 above)
Comment (loevgaard) [reply src/Stack/ElementStack.php:54]: I know that `Page` is also an `ElementInterface`, but pages are not added to the element stack, but maybe that would be the consistent thing to do. Right now the element stack is only used by the toolbar

File: src/Renderer/ElementId.php (hunk 4 above)
Comment (loevgaard) [reply src/Renderer/ElementId.php:57]: We don't use pages the same way we use the other elements. I will add it when we need or when someone needs it :)

File: src/Stack/ElementStack.php (hunk 6 above)
Comment (loevgaard) [reply src/Stack/ElementStack.php:54]: Same answer as the other page related question :)

File: src/Resources/views/view/non_existing_template.html.twig (hunk 5 above)
Comment (loevgaard) [reply src/Resources/views/view/non_existing_template.html.twig:4]: These were here before, but I have added todos

**Pull request merged**